src/components/Wizards/CreateVmWizard/CreateVmWizard.js

```diff
@@ -156,50 +170,64 @@ export class CreateVmWizard extends React.Component {
       title: 'Namespace',
       type: 'dropdown',
       default: '--- Select Namespace ---',
-      values: () => this.getValueFromState('namespaces'),
+      values: () => this.props.namespaces.map(namespace => this.getName(namespace)),
       required: true
     },
     imageSourceType: {
       title: 'Provision Source',
       type: 'dropdown',
       default: '--- Select Provision Source ---',
-      values: [PROVISION_SOURCE_PXE, PROVISION_SOURCE_URL, PROVISION_SOURCE_REGISTRY],
+      values: [PROVISION_SOURCE_PXE, PROVISION_SOURCE_URL, PROVISION_SOURCE_REGISTRY, PROVISION_SOURCE_TEMPLATE],
       required: true
     },
     registryImage: {
       title: 'Registry Image',
       required: true,
-      isVisible: basicVmSettings => get(basicVmSettings, 'imageSourceType.value') === PROVISION_SOURCE_REGISTRY
+      isVisible: basicVmSettings => this.isImageSourceType(basicVmSettings, PROVISION_SOURCE_REGISTRY)
```

Comment:
I don't think this will work.
The `isVisible` is a function  with returns a function.
Looking at actual calling of the `isVisible()` (lines 89, 108), just the first function is called, so always evaluated to `true`.

It can be good-enough to just:
```
isImageSourceType = type => (basicVmSettings => get(basicVmSettings, 'imageSourceType.value') === type)
isVisible: this.isImageSourceType(PROVISION_SOURCE_REGISTRY)
```

Reply:
Sorry for confusion, the call is as expected.